packages/fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -115,13 +108,13 @@ pub fn generate_script_main_function(
         #doc
         pub fn #name(&self #(,#arg_declarations)*) -> ScriptCallHandler<#output_type> {
             let arg_name_tokens = [#(#arg_names.into_token()),*];
-            let script_data = ABIEncoder::encode(&arg_name_tokens).expect("Cannot encode script
-            arguments").resolve(0);
             let script_binary = std::fs::read(self.binary_filepath.as_str())
                                         .expect("Could not read from binary filepath");
+            let script_offset = get_base_script_offset() + script_binary.len();
```

Comment:
Ok that's the catch. How did you realize we were missing a base script offset?

Reply:
When we were looking at the various offsets that we found manually I made a mistake and divided them by 8. I wanted to make  bytes but the offset was already bytes. Then I subtracted the script binary length and got weird results.. When I did it without the division I stared getting constant offsets and later found that it was the base offset above ^

Reply:
I initially wanted to turn the offset to bytes because I thought that the offset we got was way to big.. later @segfault-magnet told me that contract calls also have around a 10kB offset, we checked it and then it all clicked :D